crates/sui-graphql-rpc/src/types/object.rs

```diff
@@ -32,12 +32,25 @@ use sui_types::object::{
 #[derive(Clone, Debug)]
 pub(crate) struct Object {
     pub address: SuiAddress,
+    pub state: ObjectState,
 
     /// Representation of an Object in the Indexer's Store.
     pub stored: Option<StoredObject>,
+}
 
-    /// Deserialized representation of `stored_object.serialized_object`.
-    pub native: NativeObject,
+#[derive(Clone, Debug)]
+pub(crate) enum ObjectState {
```

Comment:
I like this development!  I think you should take it even further though, namely it feels like the `stored` field also belongs in this enum, and you need the following cases:

```rust
pub(crate) enum ObjectKind {
  NotIndexed(NativeObject),
  Live(NativeObject, StoredObject),
  Historic(NativeObject, StoredHistoryObject),
  WrappedOrDeleted,
  OutsideAvailableRange,
}
```

Note the name changes
- from `ConsistentRange` to `AvailableRange` to match what we call it in the GraphQL API.
- from `ObjectState` to `ObjectKind` (this is a common Rust idiom, so it's nice to name it to match that idiom.  I know we have another enum called `ObjectKind` at the moment, but that's going away soon).